enhancements/metering/user-provided-pricesheets.md

```diff
@@ -0,0 +1,205 @@
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+TBD
+
+## Summary
+
+User provided price sheets is a way for users to define custom price rates for calculating cost reports on resource reservations or usage.
+It is an opt-in feature that would require the user to provide configuration that defines the prices and rates they desire.
+
+## Motivation
+
+Currently metering only supports calculating costs using AWS billing data, but does not support calculating costs for other clouds nor on-premise solutions.
+User provided price sheets would allow users to define their own prices for uses in clouds or environments which don't yet have native support from Metering.
+
+Additionally, by creating the concept of price sheets, we must define our cost report queries to work against a well defined API.
+Instead of writing a cost report query per cloud provider, we can write queries that will work against any price sheet.
+This would allow us, and users to programmatically generate price sheets, more easily unlocking support cost reports on different clouds and on-premise environments.
+
+### Goals
+
+- Allow users to define their own prices for resources, eg: $.015 per core hour.
+- Enable better use of cost [ReportQueries][reportqueries] by standardizing cost calculations to use price sheet Presto tables
+
+### Non-Goals
+
+- Perfect ease-of use. Price sheets builds upon multiple core concepts already available in Metering. We can accomplish everything we need with existing functionality, but it may not be as easy as it could be. We will add higher level abstractions to hide the lower level details once the concept is proven.
+- Adding support for new clouds. Supporting user provided price sheets is the primary goal, and the secondary goal is ensuring cloud bills could be used to generate a price sheet.
+
+## Proposal
+
+Currently, cost calculations for AWS are done by determining the total cluster cost for the reporting period by looking at the AWS billing data, using node metrics to determine the nodes active during the reporting period.
+This total cluster cost was then split up by pod, or namespace by calculating the percentage of usage or request for the pod/namespaces in terms of CPU or memory.
+This percentage was then multiplied by the cost for the reporting period to get the total cost for a pod or namespace.
+
+The price sheet model changes how we calculate cost, but actually provides more granularity and actually translates to the same concepts as above but with a bit more complexity as a result of the flexibility.
+
+Calculating cost with a price sheet is about having a price for a rate of usage and multiplying the usage by that price, much like the way we do cost reporting with AWS today.
```

Comment:
Today, with AWS, to calculate cost, we do cost reporting by multiplying **usage** by a **price for rate of usage**. We propose to do this for other environments by means of a price sheet.